pkg/network/amqp.go

```diff
@@ -86,8 +86,8 @@ func (a *Amqp) PublishPersistentMessage(exchange, key string, body []byte) error
 }
 
 // OnMessage receive messages and put them on channel
-func (a *Amqp) OnMessage(msgChan chan InMsg, queueName, exchangeName, key string) error {
-	err := a.declareExchange(exchangeName)
+func (a *Amqp) OnMessage(msgChan chan InMsg, queueName, exchangeName, kind, key string) error {
+	err := a.declareExchange(exchangeName, kind)
```

Comment:
What do you think about using the same naming convention across identifiers (`exchangeType` in favor of `kind`)?  Curiously, it seems [streadway](https://github.com/streadway/amqp/blob/master/_examples/simple-producer/producer.go#L17)  also uses this standard :smiley:

In addition, don't you think `queueName` seems redundant? Here's [another example from streadway](https://github.com/streadway/amqp/blob/master/_examples/simple-consumer/consumer.go). 

```suggestion
func (a *Amqp) OnMessage(msgChan chan InMsg, queue, exchangeName, exchangeType, key string) error {
	err := a.declareExchange(exchangeName, kind)
```

Reply:
Done, thanks!

[streadway](https://github.com/streadway/amqp/blob/55cd81a10b0c3b411968dfeb3cdd2adb4cb975a7/channel.go#L1140) uses `kind` as `channel.ExchangeDeclare()` parameter but I think your suggestion fits well in our code.

`queueName` doesn't seems redundant in general but as the function scope is small, I agree with your suggestion.